File: sessions_mount_dir/1pyu7bdgsag60xjqxilg1fcq9/frontend/src/api/courses.js
import axios from 'axios';

const API_BASE_URL = '/api/v1';

/**
 * 获取所有课程列表
 * @returns {Promise<Array>} 课程列表
 */
export const getCourses = async () => {
  try {
    const response = await axios.get(`${API_BASE_URL}/courses`);
    return response.data;
  } catch (error) {
    console.error('获取课程列表失败:', error);
    throw error;
  }
};

/**
 * 获取单个课程详情
 * @param {number} id 课程ID
 * @returns {Promise<Object>} 课程详情
 */
export const getCourseById = async (id) => {
  try {
    const response = await axios.get(`${API_BASE_URL}/courses/${id}`);
    return response.data;
  } catch (error) {
    console.error('获取课程详情失败:', error);
    throw error;
  }
};

/**
 * 创建新课程预约
 * @param {Object} courseData 课程数据
 * @returns {Promise<Object>} 创建的课程
 */
export const createCourse = async (courseData) => {
  try {
    const response = await axios.post(`${API_BASE_URL}/courses`, courseData);
    return response.data;
  } catch (error) {
    console.error('创建课程失败:', error);
    throw error;
  }
};

/**
 * 更新课程信息
 * @param {number} id 课程ID
 * @param {Object} courseData 更新的课程数据
 * @returns {Promise<Object>} 更新后的课程
 */
export const updateCourse = async (id, courseData) => {
  try {
    const response = await axios.put(`${API_BASE_URL}/courses/${id}`, courseData);
    return response.data;
  } catch (error) {
    console.error('更新课程失败:', error);
    throw error;
  }
};

/**
 * 删除课程
 * @param {number} id 课程ID
 * @returns {Promise<void>}
 */
export const deleteCourse = async (id) => {
  try {
    await axios.delete(`${API_BASE_URL}/courses/${id}`);
  } catch (error) {
    console.error('删除课程失败:', error);
    throw error;
  }
};

/**
 * 获取教练排班信息
 * @returns {Promise<Array>} 教练排班列表
 */
export const getCoachSchedules = async () => {
  try {
    const response = await axios.get(`${API_BASE_URL}/coaches`);
    return response.data;
  } catch (error) {
    console.error('获取教练排班失败:', error);
    throw error;
  }
};

File: sessions_mount_dir/00ydft4j0yyj8dl4aqiyiwh90/backend/app/routers/courses.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List

from ..database import get_db
from ..models import Course
from ..schemas import CourseSchema, CourseCreate

router = APIRouter(
    prefix="/api/v1/courses",
    tags=["courses"]
)

@router.get("/", response_model=List[CourseSchema])
def get_courses(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    """
    获取课程列表
    
    Args:
        skip: 跳过的记录数
        limit: 返回的最大记录数
        db: 数据库会话
        
    Returns:
        课程列表
    """
    courses = db.query(Course).offset(skip).limit(limit).all()
    return courses

@router.post("/", response_model=CourseSchema)
def create_course(course: CourseCreate, db: Session = Depends(get_db)):
    """
    创建新课程
    
    Args:
        course: 课程创建数据
        db: 数据库会话
        
    Returns:
        创建的课程
    """
    db_course = Course(**course.dict())
    db.add(db_course)
    db.commit()
    db.refresh(db_course)
    return db_course

@router.get("/{course_id}", response_model=CourseSchema)
def get_course(course_id: int, db: Session = Depends(get_db)):
    """
    获取单个课程详情
    
    Args:
        course_id: 课程ID
        db: 数据库会话
        
    Returns:
        课程详情
    """
    course = db.query(Course).filter(Course.id == course_id).first()
    if not course:
        raise HTTPException(status_code=404, detail="Course not found")
    return course

@router.put("/{course_id}", response_model=CourseSchema)
def update_course(course_id: int, course: CourseCreate, db: Session = Depends(get_db)):
    """
    更新课程信息
    
    Args:
        course_id: 课程ID
        course: 课程更新数据
        db: 数据库会话
        
    Returns:
        更新后的课程
    """
    db_course = db.query(Course).filter(Course.id == course_id).first()
    if not db_course:
        raise HTTPException(status_code=404, detail="Course not found")
    
    for key, value in course.dict().items():
        setattr(db_course, key, value)
    
    db.commit()
    db.refresh(db_course)
    return db_course

@router.delete("/{course_id}")
def delete_course(course_id: int, db: Session = Depends(get_db)):
    """
    删除课程
    
    Args:
        course_id: 课程ID
        db: 数据库会话
        
    Returns:
        操作结果
    """
    course = db.query(Course).filter(Course.id == course_id).first()
    if not course:
        raise HTTPException(status_code=404, detail="Course not found")
    
    db.delete(course)
    db.commit()
    return {"message": "Course deleted successfully"}